### PR TITLE
FEATURE: Add an interface for an asset content extractor service

### PR DESCRIPTION
**Pull request by daniellienert**
Also adds a DTO to transfer the result

### CODE REVIEW

File: Classes/Dto/AssetContent.php [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ */
+
+class AssetContent
+{
+
+    /**
+     * @var string
+     */
+    protected $content;
+    /**
```
Comment (andrehoffmann30) [inline Classes/Dto/AssetContent.php:23]: One blank line is missing here and for the following property definitions.

File: Classes/Eel/IndexingHelper.php [hunk 2]
```diff
@@ -11,8 +11,10 @@
  * source code.
  */
 
-use Neos\Eel\ProtectedContextAwareInterface;
+
```
Comment (andrehoffmann30) [inline Classes/Eel/IndexingHelper.php:14]: This blank line can be deleted

**Comment by kitsunet** [comment]
@daniellienert This breaks compatibility with simple search (or any other search implementation for that matter) because they do not have an asset extraction implemention :(

**Comment by hvdhotline** [comment]
(Cause?) What I get is a "The object "Neos\ContentRepository\Search\AssetExtraction\AssetExtractorInterface" which was specified as a property in the object configuration of object "Neos\ContentRepository\Search\Eel\IndexingHelper" (automatically registered class) does not exist."-Error :-(

**Comment by daniellienert** [comment]
Hey @kitsunet, 
Sorry, I don't get it.
1. There has been a method in the indexing helper, I just renamed the method and tightened the interface (https://github.com/neos/content-repository-search/pull/36/files#diff-5f8995043335ba0da26853f676ab56710117614ac126bc145904cba34498e6a6R211)
2. It was released in a major version, which indicates there are breaking changes. Why does simplesearch pull this version, it requires ^3.0 as stable version.
3. The previous implementation added the asset content as base64, which only made sense for the old Elasticsearch versions with special plugins available. Does this work with Simplesearch. With the extractor interface implementations I indeed had simple search in mind where you could easily build an extractor using TIKA

**Comment by daniellienert** [comment]
@hvdhotline, ah I get it. So it would help to use the object manager instead of injecting the interface. But then again, why do you get that code as there is no simplesearch version which is compatible to 4.0 of this package.

**Comment by hvdhotline** [comment]
@daniellienert (Sorry first ...) I installed the "depending" Packages via composer directly to just have a look what happens and might work and what I perhaps could contribute/learn. _(It's just the SimpleSearch.ContentRepositoryAdaptor in a cloned version)_. Thus I stumbled upon that **Error, which I can't understand ... cause the php-code is there** and exist. And I was looking for a Hint to understand. But perhaps that's too much flow/php-concepts – (where is a good start to read sth about "object manager instead of injecting the interface"? :-)

**Pull request merged**